bridge/pier/syncer.go

```diff
@@ -212,17 +224,41 @@ func (syncer *Syncer) startSubscription(ctx context.Context, subscription ethere
 func (syncer *Syncer) processHeader(newHeader *types.Header) {
 	syncer.Logger.Debug("New block detected", "blockNumber", newHeader.Number)
 
-	latestNumber := newHeader.Number
+	// adding into queue
+	syncer.headerQueue.PushBack(&LightHeader{
+		Number: newHeader.Number,
+		Time:   newHeader.Time,
+	})
+
+	// current time
+	currentTime := uint64(time.Now().UTC().Unix())
+	confirmationTime := uint64(helper.GetConfig().TxConfirmationTime.Seconds())
```

Comment:
Can probably keep it in Syncer struct during `NewSyncer` to avoid extra function calls every time.

Reply:
Ah yeah. Agree.